### PR TITLE
Add support for Arilux AL-C0x Wifi RGB/RGBW Controller

### PR DESCRIPTION
**Pull request by ducseb**
Led On/Led Off/Rgb Command

### CODE REVIEW

**Comment by corbinmunce** [comment]
Hi,
I recently changed the Xiaomi Gateway code in HardwareController.js, can you update this in your pull request?  Needed to add password and network configuration.
Thanks,
Corbin

**Comment by ducseb** [comment]
Hi,
Ok no problem.
Regards.

**Comment by corbinmunce** [comment]
Thank you

File: main/RFXNames.cpp [hunk 1]
```diff
@@ -2759,6 +2760,11 @@ bool GetLightCommand(
 			cmd=Limitless_SetRGBColour;
```
Comment (gizmocuz) [inline main/RFXNames.cpp:2760]: Can't we use the 'Set Colour' command ? Yes, we might have to change the 'value' member of the _tLimitlessLights structure to an integer for this

File: main/RFXNames.cpp [hunk 2]
```diff
@@ -2759,6 +2760,11 @@ bool GetLightCommand(
...
 			return true;
 		}
+		else if (switchcmd == "Set HexRGBColor")
+		{
+			cmd = Limitless_SetHEXColour;
```
Comment (gizmocuz) [inline main/RFXNames.cpp]: Limitless_SetHEXColour is added by 'someone' but not used (also not needed as we can use 'Limitless_SetRGBColour'
I see internally you also use 'RGB' in the variable/function
(and i think also pLed->Value which is a uint8 i thought

**Review by gizmocuz — COMMENTED** [review]
Thanks for your great work.
A few comments

File: main/RFXNames.cpp (hunk 2 above)
Comment (ducseb) [reply main/RFXNames.cpp]: I have deleted mention to Limitless_SetHEXColour  :)

My HSV->RGB conversion is based on MySensorBase.cpp/BleBox.cpp/ZWaveBase.cpp
Do you want me to change something ?

**Comment by gizmocuz** [comment]
In the end... all hardware should use native RGB/Hex colors, but currently this is not so
That will be a major change, so we get rid of the HSV/L stuff, because also the zwave hardware supports direct RGB values
To keep it all compatible now, i would suggest to use the current limitedled implementation with the Hue stuff
In the SendSwitchCommand (mainworker) you have total control what you would like to send to the hardware
Or store the level internally in your class, and do a combination with the Hue when setting the color
Hope thats an option ?

**Comment by ducseb** [comment]
OK, thanks for the information and advices :).
I have committed the fix for "Limitless_SetHEXColour".

This hardware plugin use the limitedled implementation like other RGB controller (i used Yeelight plugin as a source of inspiration)
For now, On/Off and basic rgb color is working, do you think it's possible to add this pull request ?

I'll try to implement the level storing later tonight or tomorrow.

File: main/RFXNames.cpp [hunk 3]
```diff
@@ -2758,7 +2759,7 @@ bool GetLightCommand(
 		{
 			cmd=Limitless_SetRGBColour;
 			return true;
-		}
+		}
```
Comment (gizmocuz) [inline main/RFXNames.cpp]: could you undo this change (extra tab)

File: main/WebServer.cpp [hunk 4]
```diff
@@ -1045,7 +1048,7 @@ namespace http {
 				(htype == HTYPE_RFXLAN) || (htype == HTYPE_P1SmartMeterLAN) || (htype == HTYPE_YouLess) || (htype == HTYPE_RazberryZWave) || (htype == HTYPE_OpenThermGatewayTCP) || (htype == HTYPE_LimitlessLights) ||
 				(htype == HTYPE_SolarEdgeTCP) || (htype == HTYPE_WOL) || (htype == HTYPE_ECODEVICES) || (htype == HTYPE_Mochad) || (htype == HTYPE_MySensorsTCP) || (htype == HTYPE_MySensorsMQTT) || (htype == HTYPE_MQTT) || (htype == HTYPE_FRITZBOX) ||
 				(htype == HTYPE_ETH8020) || (htype == HTYPE_Sterbox) || (htype == HTYPE_KMTronicTCP) || (htype == HTYPE_SOLARMAXTCP) || (htype == HTYPE_SatelIntegra) || (htype == HTYPE_RFLINKTCP) || (htype == HTYPE_Comm5TCP) || (htype == HTYPE_CurrentCostMeterLAN) ||
-				(htype == HTYPE_NefitEastLAN) || (htype == HTYPE_DenkoviSmartdenLan) || (htype == HTYPE_Ec3kMeterTCP) || (htype == HTYPE_MultiFun) || (htype == HTYPE_ZIBLUETCP)
+				(htype == HTYPE_NefitEastLAN) || (htype == HTYPE_DenkoviSmartdenLan) || (htype == HTYPE_Ec3kMeterTCP) || (htype == HTYPE_MultiFun) || (htype == HTYPE_ZIBLUETCP)
```
Comment (gizmocuz) [inline main/WebServer.cpp]: could you undo this change (extra space)

File: main/WebServer.cpp [hunk 5]
```diff
@@ -1367,7 +1373,7 @@ namespace http {
 				(htype == HTYPE_MySensorsTCP) || (htype == HTYPE_MySensorsMQTT) || (htype == HTYPE_MQTT) || (htype == HTYPE_FRITZBOX) || (htype == HTYPE_ETH8020) || (htype == HTYPE_Sterbox) ||
 				(htype == HTYPE_KMTronicTCP) || (htype == HTYPE_SOLARMAXTCP) || (htype == HTYPE_SatelIntegra) || (htype == HTYPE_RFLINKTCP) ||
 				(htype == HTYPE_Comm5TCP || (htype == HTYPE_CurrentCostMeterLAN)) ||
-				(htype == HTYPE_NefitEastLAN) || (htype == HTYPE_DenkoviSmartdenLan) || (htype == HTYPE_Ec3kMeterTCP) || (htype == HTYPE_MultiFun) || (htype == HTYPE_ZIBLUETCP)
+				(htype == HTYPE_NefitEastLAN) || (htype == HTYPE_DenkoviSmartdenLan) || (htype == HTYPE_Ec3kMeterTCP) || (htype == HTYPE_MultiFun) || (htype == HTYPE_ZIBLUETCP)
```
Comment (gizmocuz) [inline main/WebServer.cpp]: could you undo this change (extra space)

**Review by gizmocuz — COMMENTED** [review]
Sure no problem, only please correct these two anomalies

File: main/RFXNames.cpp (hunk 3 above)
Comment (ducseb) [reply main/RFXNames.cpp]: Done.

**Comment by gizmocuz** [comment]
Thanks for the great work!

**Comment by gizmocuz** [comment]
Hi, i wonder if you tested the hardware controller , fixed in
https://github.com/domoticz/domoticz/commit/6864eb276e44d8ef6f9846ac521ab04c1678edd7
